File: c++/198_house_robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        
        if(nums.empty()) return 0;
        
        for(int i = 2; i < nums.size(); i++){
            int maxnum = 0;
            for(int j = 0; j < i-1; j++)
                maxnum = max(nums[j], maxnum);
            nums[i] += maxnum;
            
        }
        return *max_element(nums.begin(), nums.end());
        
    }
};

File: 198_house_robber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n == 0:
            return 0
        
        for i in range(2, n):
            maximum = 0
            for j in range(i-1):
                maximum = max(nums[j], maximum)
            nums[i] += maximum
            
        return max(nums)
                